optimization/ht_functions.py

```diff
@@ -1,12 +1,128 @@
+    """
+
+    #From Li, Seem, and Li, "IRJ, 
+    #"A New Explicity Equation for Accurate Friction Factor 
+    # Calculation for Smooth Tubes" 2011
+    f_fd=(-0.001570232/math.log(Re) +
+           0.394203137/math.log(Re)**2 +
+           2.534153311/math.log(Re)**3) * 4 
+    
+    if relrough > 1e-5:
+        #Offor and Alabi, 
+        #Advances in Chemical Engineering and Science, 2016, 6, 237-245
+        f_fd=(-2*math.log(
+                 (relrough/3.71) -
+                 (1.975/Re) * math.log((relrough/3.93)**1.092 +
+                 (7.627/(Re + 395.9)) , 10))) ** (-2)
+    
+    #Gnielinski, V.,, Int. Chem. Eng., 16, 359, 1976
+    Nusselt_L= ((f_fd/8)*(Re-1000)*Pr)/(1+12.7*math.sqrt(f_fd/8)*(Pr **(2/3) - 1)) 
+    
+    if (Pr<0.5):
+        # Notter and Sleicher, Chem. Eng. Sci., Vol. 27, 1972
+        Nusselt_L_lp =4.8 + 0.0156 * Re**0.85 * Pr**0.93
+        if (Pr<0.1):
+            Nusselt_L = Nusselt_L_lp
+        else:
+            Nusselt_L = Nusselt_L_lp+(Pr-0.1)*(Nusselt_L-Nusselt_L_lp)/0.4
+
+    #account for developing flow
+    f=f_fd*(1+(1/LD)**0.7) 
+    Nusselt_L*=(1+(1/LD)**0.7)
+    
+    return Nusselt_L, f
+
+def pipeflow_laminar(Re, Pr, LD, relrough):
+    """Laminar pipeflow correlation from EES. This function provides a nusselt
+    number and friction factor for laminar flow in a pipe.
+
+    Arguments:
+    ----------
+        Re (float): flow Reynold's numbe [-]
+        Pr (float): flow Prandtl number [-]
+        LD (float): length over diameter [-]
+        relrough (float) : relative roughness [-]
+    
+    Returns:
+    --------
+        Nusselt_T (float): nusselt number at constant temperature
+        Nusselt_H (float): nusselt number at constant heat flux
+        f (float): friction factor
+    """
+
+    Gz = Re* Pr /LD     
+    x = LD / Re
+    fR = 3.44 / math.sqrt(x) +\
+        (1.25/(4*x) + 16 - 3.44/math.sqrt(x)) /\
+        (1 + 0.00021 * x**(-2))
+
+    f = 4 * fR / Re
+    Gm = Gz**(1/3)
+    Nusselt_T = 3.66 + ((0.049+0.02/Pr)*Gz**1.12)/(1+0.065*Gz**0.7)
+    Nusselt_H = 4.36 + ((0.1156 +0.08569 /Pr**0.4)*Gz)/(1+0.1158*Gz**0.6) 
+    
+    return Nusselt_T, Nusselt_H, f
+    
+def pipeflow_nd(Re, Pr, LD, relrough):
+    """Nusselt correlation procedure from EES. This function provides a nusselt
+    number and friction factor for flow in a pipe. It accounts for laminar,
+    turbulent and transistional flow.
+    
+    Arguments:
+    ----------
+        Re (float): flow Reynold's numbe [-]
+        Pr (float): flow Prandtl number [-]
+        LD (float): length over diameter [-]
+        relrough (float) : relative roughness [-]
+    
+    Returns:
+    --------
+        Nusselt_T (float): nusselt number at constant temperature
+        Nusselt_H (float): nusselt number at constant heat flux
+        f (float): friction factor
+    """
+    if Re > 3000:
+        Nusselt_T, f = pipeflow_turbulent(Re, Pr, LD, relrough)
+        Nusselt_H = Nusselt_T
```

Comment:
I don't like that, I would prefere if both pipe flow_xxx return all of `Nusselt_T`, `Nusselt_H`, `f`... (with `Nusselt_H = Nusselt_T` for the turbulent case.